lib/mongodb-filter-wrapper.js

```diff
@@ -0,0 +1,155 @@
+		const modelFields = model.constructor.fields || {};
+
+		return this.getParsedFilters(filters, modelFields);
+	}
+
+	/**
+		* Validates filters given and loop calling to parse
+		* @param filters {Object} Filters defined as parameters
+		* @param modelFields {Object} Model that will use the filter
+		* @returns {Array|Object} The filter parsed to use in MongoDB
+		*/
+	static getParsedFilters(filters, modelFields) {
+		const orCondition = Array.isArray(filters);
+
+		let parsedFilters = orCondition ? [] : {};
+
+		// Only arrays as a filter
+		if(filters !== undefined && typeof filters === 'object' && Object.keys(filters).length > 0 && !Array.isArray(filters))
+			filters = [filters];
+
+		if(filters && filters.length > 0) {
+			// First loop over all filters given by parameters
+			filters.forEach(filterGiven => {
+				const fullyParsedFilter = this.parseObjectFilter(filterGiven, modelFields);
+				parsedFilters = (orCondition && Array.isArray(parsedFilters))
+					? [...parsedFilters, fullyParsedFilter] : Object.assign(parsedFilters, fullyParsedFilter);
+			});
+		}
+
+		if(orCondition && parsedFilters.length > 0) {
+			return {
+				$or: parsedFilters
+			};
+		}
+
+		return parsedFilters;
+	}
+
+	/**
+		* Parse the filter comparing given fields name and model fields
+		* @param filters {Object} One filter given with her type and value(only uses that info)
+		* @param modelFields {Object} Model that will use the filter if exists
+		* @returns {Object} The filter parsed with all the values by one
+		*/
+	static parseObjectFilter(filters, modelFields) {
+		const filterToResponse = {};
+
+		for(const [field, value] of Object.entries(filters)) {
+			const fieldName = this.getFieldToCompare(field, modelFields);
+
+			const filter = (typeof value !== 'object') ? { value } : value;
+
+			filterToResponse[fieldName] = this.createFilter(filter, modelFields[field]);
+		}
+		return filterToResponse;
+	}
+
+	/**
+		* Creates the object or the value(string/array) to use in the filter
+		* @param filter {Object} Filter given as parameter to parse with type and value
+		* @param modelFields {Object} Model field that will use the filter, if exists
+		* @returns {string|Object} Value to use in filter, with type or if is an equal only the value to
+		* search
+		*/
+	static createFilter(filter, modelFields) {
+		const type = this.getFilterType(filter, modelFields);
+		const value = this.getValue(filter);
+
+		if(type === 'equal')
+			return value;
+
+		return {
+			[this.filtersAllowed[type]]: value
+		};
+	}
+
+	/**
+		* Gets the value to use in the filter, if is a object(suppose ID) transform to string
+		* else search that exists the key value to return, or only returns an empty string
+		* @param filter {Object} With all the filter
+		* @returns {string|number|object|array} With the value to filter
+		*/
+	static getValue(filter) {
+		let value = '';
+
+		if(typeof filter === 'object' && filter.value === undefined)
+			value = filter.str;
```

Comment:
```suggestion
			value = filter.toString();
```
La propiedad `str` no existe en `ObjectID` de mongo, entiendo que en el mock para los tests si **:shrek:** pero utilizando `toString()` funciona como se espera para ambos casos, tanto el real como el mock.